oci/cache/image.go

```diff
@@ -30,9 +30,11 @@ func RemoteImage(ref name.Reference, options ...remote.Option) (v1.Image, error)
 		goto lookup
 	}
 
+	imageCacheMu.Lock()
 	if image, ok := imageCache[name]; ok {
 		return image, nil
 	}
+	imageCacheMu.Unlock()
```

Comment:
This is wrong right? What if the `if` returns and the mutex is never unlocked it would make everything go boom

Either unlock in the return also, or use a `sync.Map`